src/cli/config.rs

```diff
@@ -0,0 +1,181 @@
+//! Global config for the CLI
+//! Includes settings for cache locations and proof backend
+use std::collections::HashMap;
+
+use crate::config::{lurk_config_file, Settings, LURK_CONFIG};
+use crate::field::LanguageField;
+use camino::Utf8PathBuf;
+use config::{Config, ConfigError, Environment, File};
+use once_cell::sync::OnceCell;
+use serde::Deserialize;
+
+use super::backend::Backend;
+use super::paths::{circom_default_dir, commits_default_dir, proofs_default_dir};
+
+/// Global config varable for `CliSettings`
+pub static CLI_CONFIG: OnceCell<CliSettings> = OnceCell::new();
+
+/// Gets the `CLI_CONFIG` settings. If uninitialized, sets the global variable
+/// in the following order (greatest to least precedence):
+/// - `settings` map if provided, e.g. with key ("proofs", "$HOME/lurk-rs/proofs")
+///   This contains any CLI args, set e.g. by `lurk --proofs-dir /path/to/proofs_dir`
+/// - Env var per setting, e.g. `LURK_PROOFS_DIR`
+/// - Config file, which also has a configurable location (see `lurk_config_file()`),
+///   and has the following syntax for e.g. TOML:
+///   ```toml
+///   proofs_dir = "/path/to/proofs_dir"
+///   ```
+///   Other file formats are supported by the `config` crate, but only TOML is tested
+/// - Default values, e.g. `$HOME/.lurk/proofs`
+pub fn cli_config(
+    config_file: Option<&Utf8PathBuf>,
+    settings: Option<&HashMap<&str, String>>,
+) -> &'static CliSettings {
+    LURK_CONFIG
+        .set(Settings::from_config(lurk_config_file(config_file), settings).unwrap_or_default())
+        .unwrap_or(());
+    CLI_CONFIG.get_or_init(|| {
+        CliSettings::from_config(lurk_config_file(config_file), settings).unwrap_or_default()
+    })
+}
+
+/// Contains the CLI configuration settings
+// NOTE: Config settings share the same file for both the Lurk library and the CLI.
+// It's good practice to avoid duplication of shared settings like `public_params_dir`
+// in downstream configs like these to prevent conflicts.
```

Comment:
This is a bit confusing, as a user-specified `rc` and `limit` value in the config file will affect the CLI but not the Lurk library or benchmarks as one might expect. It may be more user-friendly to have a separate `cli.toml` rather than sharing settings with `lurk.toml`, or better yet support a table-based TOML structure like the following:
```toml
[lurk]
public_params_dir = "/path/to/public_params"

[cli]
proofs_dir = "/path/to/proofs"
rc = 100
limit = 100000000

Reply:
I think this deserves its own issue.

Reply:
https://github.com/lurk-lab/lurk-rs/issues/793